instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -693,5 +695,8 @@ async def otel_send(message):
                         pass
 
                 await send(message)
+            if message["type"] == "http.response.body":
```

Comment:
Small thing which a coworker pointed out; the request event could indicate that trailers are expected after the request body: https://asgi.readthedocs.io/en/latest/specs/www.html#response-start-send-event:~:text=If%20more_body%20is%20set%20to%20False%2C%20and%20the%20server%20is%20not%20expecting%20Response%20Trailers%20this%20will%20complete%20the%20response. In that case, I think you would want to wait for trailers to be sent before ending the span

Reply:
In my latest commit(6a2ff51) I added support for HTTP trailers according to the ASGI spec. The drawback of supporting this is of course having to maintain state throughout the request. I currently maintain the state through a nonlocal variable. I'm open to suggestions on more elegant ways to track the state as I don't find the current method very elegant.

However I do think some discussion is warranted around HTTP trailers because AFAIK there is no publicly available ASGI server nor framework which conforms to the spec / supports HTTP trailers. So I'm not exactly sure the instrumentation should support something that nobody seems to use at the cost of increasing complexity.

Reply:
The closure solution looks pretty clean to me, I don't have any issue with it. I agree this may be unimportant in practice, but nice to have for completeness since it's only a few extra lines